ui_vue/cypress/support/commands.ts

```diff
@@ -1,39 +1,87 @@
 /// <reference types="cypress" />
-// ***********************************************
-// This example commands.ts shows you how to
-// create various custom commands and overwrite
-// existing commands.
-//
-// For more comprehensive examples of custom
-// commands please read more here:
-// https://on.cypress.io/custom-commands
-// ***********************************************
-//
-//
-// -- This is a parent command --
-// Cypress.Commands.add('login', (email, password) => { ... })
-//
-//
-// -- This is a child command --
-// Cypress.Commands.add('drag', { prevSubject: 'element'}, (subject, options) => { ... })
-//
-//
-// -- This is a dual command --
-// Cypress.Commands.add('dismiss', { prevSubject: 'optional'}, (subject, options) => { ... })
-//
-//
-// -- This will overwrite an existing command --
-// Cypress.Commands.overwrite('visit', (originalFn, url, options) => { ... })
-//
-// declare global {
-//   namespace Cypress {
-//     interface Chainable {
-//       login(email: string, password: string): Chainable<void>
-//       drag(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       dismiss(subject: string, options?: Partial<TypeOptions>): Chainable<Element>
-//       visit(originalFn: CommandOriginalFn, url: string, options: Partial<VisitOptions>): Chainable<Element>
-//     }
-//   }
-// }
+
+// Import constants
+import constants from '../cypress.constants';
+
+Cypress.Commands.add('loginByGoogleApi', () => {
+  cy.visit(constants.pages.signIn.link);
+  cy.request({
+    method: constants.authApis.methods.post,
+    url: constants.authApis.google.token,
+    body: {
+      grant_type: constants.authApis.grantType.refreshToken,
+      client_id: import.meta.env.VITE_API_GOOGLE_CLIENT_ID,
+      client_secret: import.meta.env.VITE_API_GOOGLE_CLIENT_SECRET,
+      refresh_token: import.meta.env.VITE_API_GOOGLE_REFRESH_TOKEN,
+    },
+  }).then(({ body }) => {
+    const { access_token, id_token } = body;
+
+    cy.request({
+      method: constants.authApis.methods.get,
+      url: constants.authApis.google.user,
+      headers: { Authorization: `Bearer ${access_token}` },
+    }).then(({ body }) => {
+      cy.log(body);
+      const userItem = {
+        token: id_token,
+        user: {
+          googleId: body.sub,
+          email: body.email,
+          givenName: body.given_name,
+          familyName: body.family_name,
+          imageUrl: body.picture,
+        },
+      };
+  
+      window.localStorage.setItem(constants.localStorage.userDataToken, JSON.stringify(userItem));
+      cy.visit(constants.pages.dashboard.link);
+    });
+  });
+});
+
+Cypress.Commands.add('loginByGithubApi', () => {
+  cy.visit(constants.pages.signIn.link);
+
+  const clientId = import.meta.env.VITE_API_GITHUB_CLIENT_ID;
+  const clientSecret = import.meta.env.VITE_API_GITHUB_CLIENT_SECRET;
+  const code = 'd4674f99a4b3dab1c053';
```

Comment:
can you please what code used for and why is not placed in constants or env file

Reply:
this is temporary code. I have moved it to env and added comment